### DISCUSSION TITLE
Mão na Massa - Criando uma Calculadora - Soma

### DISCUSSION THREAD

**Discussion by rikemorais** [balta-io/2801#28] [Perguntas e Respostas] [answered]
Boa noite! Eu estou criando o projeto da calculador e não entendi o que pode estar acontecendo com o método Soma():

```
            float v1 = float.Parse(Console.ReadLine());
            float v2 = float.Parse(Console.ReadLine());
```

Em meu VS Code aparece um erro:

```
string? Console.ReadLine()
Reads the next line of characters from the standard input stream.

Devoluções:

The next line of characters from the input stream, or null if no more lines are available.

Exceções:

  IOException
  OutOfMemoryException
  ArgumentOutOfRangeException
```
O que causou o erro?

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde @RikeMorais, fiz um teste com seu código aqui e funcionou:
![image](https://user-images.githubusercontent.com/965305/148659860-1df9ef7e-0996-467f-938d-9663f7881d42.png)

**Comment by rikemorais** [comment]
Boa tarde mestre!

Sim, ele funciona. Eu só não entendi o motivo dos warnings em amarelo. Eu estou apenas perguntando por curiosidade.

De qualquer forma, abraço e obrigado pela atenção.

**Reply by andrebaltieri** [reply]
Ahh perfeito, não havia entendido, desculpe!
Os warnings se referem a um possível valor nulo, visto que podemo apertar enter no console sem digitar um valor.
A partir do .NET 6, todos os projetos vem com estes Warnings de nulo... você pode desabilitar isto se quiser, no seu arquivo `.csproj`, removendo a linha `<Nullable>enable</Nullable>`, porém pode ajudar bastante saber quando um valor possívelmente pode ser nulo.

Estes warnings não impedem seu programa de compilar ou rodar!

**Comment by rikemorais** [comment]
Espetacular! É disso que estou falando! Muito obrigado Boss! Super esclarecedor.

**Reply by andrebaltieri** [reply]
<3
